### PR TITLE
Projectkk2glider/lua float

### PR DESCRIPTION
**Pull request by projectkk2glider**
This PR has mixed in other double to float conversion. The main purpose of this is review and testing. 

The actual full conversion from double to float still needs some consideration, one obvious thing is GPS coordinates for which float will perhaps not be precise enough.

### CODE REVIEW

File: radio/src/thirdparty/Lua/src/lrotable.c [hunk 1]
```diff
@@ -10,77 +10,67 @@
...
-  if (strlen(name) > LUA_MAX_ROTABLE_NAME)
+  if (strlen(name) > LUA_MAX_ROTABLE_NAME) {
+    TRACE_LUA_INTERNALS("luaR_findglobal() '%s' = NAME TOO LONG", name);
     return 0;
+  }
   for (i=0; lua_rotable[i].name; i++) {
     if (!strcmp(lua_rotable[i].name, name)) {
-      *ptype = LUA_TROTABLE;
-      return i+1;
+      setrvalue(val, (void *)(size_t)(i+1));
```
Comment (projectkk2glider) [inline radio/src/thirdparty/Lua/src/lrotable.c]: This still uses ugly cast, but does not involve double of float. Not sure if we can improve it.

File: radio/src/thirdparty/Lua/src/lrotable.c [hunk 2]
```diff
@@ -60,17 +61,17 @@ luaR_result luaR_findglobal(const char * name, TValue * val) {
 
 
 luaR_result luaR_findentry(void *data, const char * key, TValue * val) {
-  unsigned idx = (unsigned)(size_t)data - 1;
+  luaR_table * table = (luaR_table *)data;
```
Comment (projectkk2glider) [inline radio/src/thirdparty/Lua/src/lrotable.c]: I think this is much nicer.

Comment (mpaperno) [inline radio/src/thirdparty/Lua/src/lrotable.c]: Definitely!

**Comment by projectkk2glider** [comment]
Replaced by #4502

**Pull request closed without merging**